### ISSUE TITLE
Module consolidation with index.d.ts files does not work

### ISSUE THREAD

**Issue by chpaulin** [aurelia/cli#626]
**I'm submitting a bug report**

* **Library Version:**
0.27.0

**Please tell us about your environment:**
* **Operating System:**
Windows 7

* **Node Version:**
6.9.4

* **NPM Version:**
3.10.10

* **Browser:**
all

* **Language:**
TypeScript 2.2

**Current behavior:**
The Aurelia-CLI build system complains that it cannot find a js file with the same name as the folder in the same path as the folder resides.

`File not found or not accessible: {path}/src/common.js. Requested by {some module}.js`

**Expected/desired behavior:**

We expected that the Aurelia-CLI would work the same way as the Typescript-skeleton does and allow us to consolidate modules in this manner.

* **What is the expected behavior?**
That we can consolidate all modules in one folder using an index.d.ts file and import all exports from the modules just importing the folder path.

index.d.ts example:
`export * from './moduleA';`
`export * from './moduleB';`
`export * from './moduleC';`

Importing all modules with one import:
`import {A, B, C} from './models'`

* **What is the motivation / use case for changing the behavior?**
Typescript supports index.d.ts file in folders that can consolidate all modules in that folder (and subfolder) into one "module". This enables us to use one import line for all exports in a folder but still separating them into separate files.
`import { PaymentDetailsModel, NextPaymentModel } from "./models";`

**Comment by chpaulin** [comment]
After some digging we figured that I hasn't worked this way before either when it comes to module loading. It works great for definition imports but not if you plan to use any logic from these imports.

Tried to create an all.ts file with the same content as the index.d.ts file in the initial post and imported it using:
`import { A, B, C } from './models/all'`

This worked! However naming it index.ts so that we could omit the /all did not work :(
